doc/build-helpers/images/dockertools.section.md

```diff
@@ -36,84 +48,216 @@ buildImage {
+    _Default value:_ `null`.
 
-The resulting repository will only list the single image `image/tag`. In the case of [the `buildImage` example](#ex-dockerTools-buildImage), it would be `redis/latest`.
+`fromImageName` _optional_
 
-It is possible to inspect the arguments with which an image was built using its `buildArgs` attribute.
+: Used to specify the image within the repository tarball in case it contains multiple images.
+  A value of `null` means that `buildImage` will use the first image available in the repository.
 
-> **_NOTE:_** If you see errors similar to `getProtocolByName: does not exist (no such protocol name: tcp)` you may need to add `pkgs.iana-etc` to `contents`.
+  :::{.note}
+  This must be used with `fromImageTag`. Using only `fromImageName` without `fromImageTag` will make `buildImage` use the first image available in the repository.
+  :::
 
-> **_NOTE:_** If you see errors similar to `Error_Protocol ("certificate has unknown CA",True,UnknownCa)` you may need to add `pkgs.cacert` to `contents`.
+    _Default value:_ `null`.
 
-By default `buildImage` will use a static date of one second past the UNIX Epoch. This allows `buildImage` to produce binary reproducible images. When listing images with `docker images`, the newly created images will be listed like this:
+`fromImageTag` _optional_
 
-```ShellSession
-$ docker images
-REPOSITORY   TAG      IMAGE ID       CREATED        SIZE
-hello        latest   08c791c7846e   48 years ago   25.2MB
-```
+: Used to specify the image within the repository tarball in case it contains multiple images.
+  A value of `null` means that `buildImage` will use the first image available in the repository.
 
-You can break binary reproducibility but have a sorted, meaningful `CREATED` column by setting `created` to `now`.
+  :::{.note}
+  This must be used with `fromImageName`. Using only `fromImageTag` without `fromImageName` will make `buildImage` use the first image available in the repository
+  :::
 
-```nix
-pkgs.dockerTools.buildImage {
-  name = "hello";
-  tag = "latest";
-  created = "now";
-  copyToRoot = pkgs.buildEnv {
-    name = "image-root";
-    paths = [ pkgs.hello ];
-    pathsToLink = [ "/bin" ];
-  };
+    _Default value:_ `null`.
 
-  config.Cmd = [ "/bin/hello" ];
-}
-```
+`copyToRoot` _optional_
 
-Now the Docker CLI will display a reasonable date and sort the images as expected:
+: Files to add to the generated image.
+  This can be either a path or a list of paths.
+  This can be seen as an equivalent of `ADD contents/ /` in a `Dockerfile`.
 
-```ShellSession
-$ docker images
-REPOSITORY   TAG      IMAGE ID       CREATED              SIZE
-hello        latest   de2bf4786de6   About a minute ago   25.2MB
-```
+  If a derivation is used in this attribute, its output path will be used instead.
+  Because of this, it's a common usage to see `pkgs.buildEnv` used in this attribute.
```

Comment:
> thoughts

Can't block on this comment, because it balloons the scope of the PR insanely. Feel free to ignore and just clarify the text without referring to anything.

Reply:
> I think we should always link to a page (e.g. in the Nix manual or nix.dev) that clarifies and disambiguates terms, as well as serve as a writing guide

I agree with this. I'm already working on the next section of `dockerTools` and encountered a similar issue when talking about Nix store objects and Nix store paths.

Given that the content related to contributing was moved out of the Nix manual, I think nix.dev is a more appropriate place to put this guide. I'll treat it as a separate piece of work not linked to this one to keep the scope of this PR focused.

Reply:
It would make sense to introduce these terms in some sort of orientation article on nix.dev, but really this is a matter of definitions, so it should find a place in the Nixpkgs manual (which prominently doesn't have a glossary yet).